### PR TITLE
[VarDumper] Add caster for pgsql resources

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

If pgsql users could give feedback on this one, that would we great!

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Ping @chanmix51

File: src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        if (-1 === $a['num rows']) {
+            foreach (self::$diagCodes as $k => $v) {
+                $a['error'][$k] = pg_result_error_field($result, $v);
+            }
+        }
+
+        $a['affected rows'] = pg_affected_rows($result);
+        $a['last OID'] = pg_last_oid($result);
+
+        $fields = pg_field_num($result);
```
Comment (chanmix51) [inline src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php]: PHP warning:  pg_field_num() expects exactly 2 parameters, 1 given

pg_field_num <> pg_num_rows ;) see http://php.net/pg_field_num

Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php]: Thanks, fixed (should be pg_num_fields)

File: src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        $a['pid'] = pg_get_pid($link);
+        $a['last error'] = pg_last_error($link);
+        $a['last notice'] = pg_last_notice($link);
+        $a['host'] = pg_host($link);
+        $a['port'] = pg_port($link);
+        $a['dbname'] = pg_dbname($link);
+        $a['options'] = pg_options($link);
+        $a['version'] = pg_version($link);
+
+        foreach (self::$linkParameters as $v) {
```
Comment (chanmix51) [inline src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php]: Access to undeclared static property: Symfony\Component\VarDumper\Caster\PgSqlCaster::$linkParameters in /home/greg/test/symfony/src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php on line 94

**Comment by chanmix51** [comment]
Nice work :+1:

File: src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php]: Oups again, fixed

**Comment by nicolas-grekas** [comment]
Cool, thank you! Do you have any suggestion to make it better?

**Comment by chanmix51** [comment]
@nicolas-grekas maybe yes. I have tested the result cast and afaicr I couldn't tell the index of the last fetched result. Postgres' cursors are scrollable, you can iterate several times on them and fetch directly the nth row of a result. I could not see that in the stub (maybe this data is not available but I think it is).

**Comment by nicolas-grekas** [comment]
I found pg_result_seek() for seeking, but nothing for getting the current cursor position. Can you find a way to get it?

**Comment by chanmix51** [comment]
Don't mind, there are no PHP functions for that, this is why it is kept in Pomm's iterators. 

All good !

File: src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        $a['last notice'] = pg_last_notice($link);
+        $a['host'] = pg_host($link);
+        $a['port'] = pg_port($link);
+        $a['dbname'] = pg_dbname($link);
+        $a['options'] = pg_options($link);
+        $a['version'] = pg_version($link);
+
+        foreach (self::$paramCodes as $v) {
+            if (false !== $v = pg_parameter_status($link, $v)) {
+                $a['param'][$v] = ;
```
Comment (sanpii) [inline src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php]: Syntax error

Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/PgSqlCaster.php]: indeed, fixed!

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**